src/Stratis.Bitcoin.Features.RPC/RpcSettings.cs

```diff
@@ -110,26 +112,29 @@ private void CheckConfigurationValidity(ILogger logger)
 
             // We can now safely assume that server was set to true in the config or that the
             // "AddRpc" callback provided a user and password implying that the Rpc feature will be used.
-            this.Server = true;
+            if (this.RpcPassword != null && this.RpcUser != null)
```

Comment:
good spot.

Reply:
To be honest I think it's strange to set server to true automatically. This means you cannot disable the server by setting it to false in the config.

Reply:
I agree, to tell the node to accept RPC command, you have to set server to true AND set the rpc fields.
I think you can remove this `this.Server = true;`.

Reply:
Done